app/views/shared/transactions/_search2.html.haml

```diff
@@ -9,7 +9,7 @@
     %fieldset.span2#calendar_filter
       - if @search.options.map(&:key).include?("date_ranges")
         %br
-        = f.input :date_range_field, collection: TransactionSearch::DateRangeSearcher.options, label: false, include_blank: false
+        = f.input :date_range_field, collection: TransactionSearch::DateRangeSearcher.options.select { |label, value| value.in?(@search_form.allowed_date_fields) }, label: false, include_blank: false
```

Comment:
Just wondering if it would be worth it to add an `only:` argument to `TransactionSearch::DateRangeSearcher.options` and pull the `select` logic in there. Might clean up the view a bit.